homeassistant/components/fan/insteon_local.py

```diff
@@ -0,0 +1,195 @@
+"""
+Support for Insteon fans via local hub control.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/fan.insteon_local/
+"""
+import json
+import logging
+import os
+from datetime import timedelta
+
+from homeassistant.components.fan import (
+    ATTR_SPEED, SPEED_OFF, SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH,
+    SUPPORT_SET_SPEED, FanEntity)
+from homeassistant.helpers.entity import ToggleEntity
+from homeassistant.loader import get_component
+import homeassistant.util as util
+
+_CONFIGURING = {}
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['insteon_local']
+DOMAIN = 'fan'
+
+INSTEON_LOCAL_FANS_CONF = 'insteon_local_fans.conf'
+
+MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(milliseconds=100)
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=5)
+
+SUPPORT_INSTEON_LOCAL = SUPPORT_SET_SPEED
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Insteon local fan platform."""
+    insteonhub = hass.data['insteon_local']
+
+    conf_fans = config_from_file(hass.config.path(INSTEON_LOCAL_FANS_CONF))
+    if len(conf_fans):
+        for device_id in conf_fans:
+            setup_fan(device_id, conf_fans[device_id], insteonhub, hass,
+                      add_devices)
+
+    else:
+        linked = insteonhub.get_linked()
+
+        for device_id in linked:
+            if (linked[device_id]['cat_type'] == 'dimmer' and
+                    linked[device_id]['sku'] == '2475F' and
+                    device_id not in conf_fans):
+                request_configuration(device_id,
+                                      insteonhub,
+                                      linked[device_id]['model_name'] + ' ' +
+                                      linked[device_id]['sku'],
+                                      hass, add_devices)
+
+
+def request_configuration(device_id, insteonhub, model, hass,
+                          add_devices_callback):
+    """Request configuration steps from the user."""
+    configurator = get_component('configurator')
+
+    # We got an error if this method is called while we are configuring
+    if device_id in _CONFIGURING:
+        configurator.notify_errors(
+            _CONFIGURING[device_id], 'Failed to register, please try again.')
+
+        return
+
+    def insteon_fan_config_callback(data):
+        """The actions to do when our configuration callback is called."""
+        setup_fan(device_id, data.get('name'), insteonhub, hass,
+                  add_devices_callback)
+
+    _CONFIGURING[device_id] = configurator.request_config(
```

Comment:
This is a very weird use case of the configurator. We shouldn't have people configure it like this, they can much better rename the devices via customize later.

Let's remove this functionality.